### DISCUSSION TITLE
Frigate

### DISCUSSION THREAD

**Discussion by mikki8** [roleoroleo/yi-hack-Allwinner-v2#959] [Q&A]
Hi Guys,

I have a bunch of Yi Pro 2K Home and running them on Frigate.
I have a Bunch of Issues also with it.
- weird glitching and ghosting in the recordings
- crashing ffmpeg(frigate says so)
- missing recording maybe due the crash
- sound delays

Even when the preview of the cam is good in frigate, home assistant and VLC player.

Is this a config issue ?
what am i doing wrong ? is there a best practice config for frigate ?

```

mqtt:
  # Optional: Enable mqtt server (default: shown below)
  enabled: true
  # Required: host name
  host: 192.168.11.163
  # Optional: port (default: shown below)
  port: 1883
  # Optional: topic prefix (default: shown below)
  # NOTE: must be unique if you are running multiple instances
  topic_prefix: frigate
  # Optional: client id (default: shown below)
  # NOTE: must be unique if you are running multiple instances
  client_id: frigate
  # Optional: user
  # NOTE: MQTT user can be specified with an environment variable or docker secrets that must begin with 'FRIGATE_'.
  #       e.g. user: '{FRIGATE_MQTT_USER}'
  user: mqtt-user
  # Optional: password
  # NOTE: MQTT password can be specified with an environment variable or docker secrets that must begin with 'FRIGATE_'.
  #       e.g. password: '{FRIGATE_MQTT_PASSWORD}'
  password: mqtt-pass

detectors:
  coral:
    type: edgetpu
    device: pci

ffmpeg:
  output_args:
    record: preset-record-generic-audio-aac
  #hwaccel_args: preset-intel-qsv-h264
  input_args: 
    -avoid_negative_ts make_zero 
    -fflags +genpts+discardcorrupt 
    -flags low_delay
    -strict experimental 
    -analyzeduration 1000M 
    -probesize 1000M 
    -rw_timeout 5000000
  hwaccel_args: 
    preset-vaapi
go2rtc:
  streams:
    cam1_low:
      - rtsp://user:pass@192.168.10.115/ch0_1.h264
    cam1:
      - rtsp://user:pass@192.168.10.115/ch0_0.h264
      - ffmpeg:cam1#audio=aac

record:
  enabled: true
  retain:
    days: 14
    mode: motion
  events:
    retain:
      default: 30
      mode: motion

cameras:
  cam1:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/cam1_low
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/cam1
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 640
      height: 360
#    snapshots:
#      enabled: true
#    record:
#      enabled: true
#      events:
#       retain:
#         default: 10
    motion:
      mask:
        - 0.284,0.011,0.29,0.096,0.027,0.113,0.008,0.023
        - 0.4,0.322,0.633,0.342,0.617,0.54,0.402,0.533
```

**Comment by roleoroleo** [comment]
Did you try to stream the cam directly instead of proxy it through go2rtc?

**Reply by mikki8** [reply]
yes i did it before the most time. migrated to go2rtc 2 months ago -  same quality - ghosting etc - but more snappier and perfomance but maybe this is a personal experience

**Comment by roleoroleo** [comment]
Are you using the last beta posted here?
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/753#issuecomment-2438422971

**Reply by mikki8** [reply]
no i have no the beta - but i will test it now

**Reply by roleoroleo** [reply]
You can wait the new release.
Few hours...

**Reply by mikki8** [reply]
yeah will update as soon its released. but set up a test cam with the firmware. i would say its snappier on the live view and better in the recordings(frame rate seams higher). didnt noticed errors and ffmpeg crashing in frigate(in 10min testing). ghosting and fragmenting is still there on movement in the frames

**Reply by roleoroleo** [reply]
You can try these changes:
- With the new firmware (0.3.3) set the option RTSP_STI=no in /tmp/sd/yi-hack/etc/system.conf
- In Frigate set the option to force ffmpeg to use rtsp_transport tcp.

**Reply by mikki8** [reply]
![grafik](https://github.com/user-attachments/assets/ba3a6a0b-0920-4e55-892f-16088573278e)
getting some errors - but i think ffmpeg is stable

i will try turn off sti later

**Reply by mikki8** [reply]
ok tested the new firmware 0.33 for now 24h
changed the "RTSP server program" to "go2rtc" on the cameras.(i dont even know if its the a new thing on this firmware. but never noticed it before.)

besides only the errors which i already posted, its running flawlessly.
i will let it for now on this settings and monitor it. but this is the it. the firmware that brings cheap cameras to frigate in a issue free experience ^^ Big Thanks
